app.js

```diff
@@ -1,18 +1,35 @@
-const express = require('express')
-const { getAllTopics, getAllEndpoints } = require('./controller')
-const app = express()
+const express = require("express");
+const {
+  getAllTopics,
+  getAllEndpoints,
+  getArticleByID,
+} = require("./controller");
+const app = express();
 
-app.get('/api/topics', getAllTopics)
+app.get("/api/topics", getAllTopics);
 
-app.get('/api', getAllEndpoints)
+app.get("/api", getAllEndpoints);
 
-app.all(('/*'), (req, res, next) => {
-  res.status(404).send({msg: 'Path Not Found'})
-  next(err)
-})
+app.get("/api/articles/:article_id", getArticleByID);
+
+app.all("/*", (req, res, next) => {
+  res.status(404).send({ msg: "Path Not Found" });
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  const errorCodes = ["22P02"];
+  if (errorCodes.includes(err.code))
+    res.status(400).send({ msg: "Bad Request" });
+  next(err);
+});
```

Comment:
You may want to do if(err.code === "22P02") here so that when more error codes come in you can give a specific error message 👍